CardanoSharp.Wallet.Test/HDWallet/DerivationTests.cs

```diff
@@ -0,0 +1,178 @@
+using System.Threading.Tasks;
+using CardanoSharp.Wallet.Enums;
+using CardanoSharp.Wallet.Extensions;
+using CardanoSharp.Wallet.Extensions.Models;
+using CardanoSharp.Wallet.Models.Derivations;
+using CardanoSharp.Wallet.Models.Keys;
+using Microsoft.AspNetCore.Cryptography.KeyDerivation;
+using Xunit;
+
+namespace CardanoSharp.Wallet.Test
+{
+    /// <summary>
+    /// HD derivation can be seen as a tree with many branches, where keys live at each node and leaf 
+    /// such that an entire sub-tree can be recovered from only a parent key (and seemingly, 
+    /// the whole tree can be recovered from the root master key).
+    /// 
+    /// <para>
+    ///     References:
+    ///     * HD Wallets: https://input-output-hk.github.io/adrestia/docs/key-concepts/hierarchical-deterministic-wallets/
+    ///     * Deriving new keys from parent keys: https://input-output-hk.github.io/adrestia/user-guide/Ed25519_BIP.pdf
+    /// </para>
+    /// </summary>
+    public class DerivationTests
+    {
+        private readonly IKeyService _keyService;
+        const string __mnemonic = "art forum devote street sure rather head chuckle guard poverty release quote oak craft enemy";
+
+        public DerivationTests()
+        {
+            _keyService = new KeyService();
+        }
+
+        [Theory]
+        [InlineData(__mnemonic)]
+        public void PathDerivationTest(string words)
+        {
+            // create two payment addresses from same root key
+            //arrange
+            var mnemonic = new KeyService().Restore(words);
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var paymentPrv1, var paymentPub1) = getKeyPairFromPath("m/1852'/1815'/0'/0/0", rootKey);
+
+            // Act
+            // Fluent derivation API
+            var derivation = testKey.Derive()                   // IMasterNodeDerivation
+                                .Derive(PurposeType.Shelley)    // IPurposeNodeDerivation
+                                .Derive(CoinType.Ada)           // ICoinNodeDerivation
+                                .Derive(0)                      // IAccountNodeDerivation
+                                .Derive(RoleType.ExternalChain) // IRoleNodeDerivation
+                                .Derive(0);                     // IIndexNodeDerivation
+
+            // Assert
+            Assert.Equal(paymentPrv1.Key, derivation.PrivateKey.Key);
+            Assert.Equal(paymentPrv1.Chaincode, derivation.PrivateKey.Chaincode);
+
+            Assert.Equal(paymentPub1.Key, derivation.PublicKey.Key);
+            Assert.Equal(paymentPub1.Chaincode, derivation.PublicKey.Chaincode);
+        }
+
+
+        [Theory]
+        [InlineData(__mnemonic, "m/1852'")]
+        [InlineData(__mnemonic, "m/1852'/1815'")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'/0")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'/0/0")]
```

Comment:
Can I start from something other than the `m`? maybe start from `Account` level?